code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm

```diff
@@ -762,6 +762,111 @@
 	create_midwife_eggs(spawncount)
 	return ..()
 
+//////////////////////////////////////////////
+//                                          //
+//           REVENANT  (GHOST)              //
+//                                          //
+//////////////////////////////////////////////
+
+/datum/dynamic_ruleset/midround/from_ghosts/revenant
+	name = "Revenant"
+	antag_datum = /datum/antagonist/revenant
+	antag_flag = "Revenant"
+	antag_flag_override = ROLE_REVENANT
+	enemy_roles = list("Security Officer", "Detective", "Head of Security", "Captain")
+	required_enemies = list(2,2,1,1,1,1,1,0,0,0)
+	required_candidates = 1
+	weight = 4
+	cost = 10
+	requirements = list(101,101,101,70,50,40,20,15,10,10)
+	repeatable = TRUE
+	var/list/spawn_locs = list()
```

Comment:
Remove this, and just make it a variable that's created in execute.

Reply:
Unless I'm missing something, there's no other way to get spawn_locs from execute to generate_ruleset_body.

Reply:
I didn't notice it was in there as well, ruleset code is weird.